### PR TITLE
Floating sidebar on desktop

### PR DESCRIPTION
**Pull request by deammer**
This PR makes the sidebar a floating element displayed on top of the content on the page.

https://user-images.githubusercontent.com/3411183/160508722-dc3d6781-398f-49fa-a5bf-1bbfac52c728.mov

### CODE REVIEW

File: src/components/local-search/SearchWrapper.tsx [hunk 1]
```diff
@@ -57,7 +57,11 @@ const SearchWrapper: FunctionComponent<Props> = ({
     }
 
     // Toggle tags in our list
-    if (newTagList.includes) updateSearchResults(filters.search, newTagList);
+    updateSearchResults(filters.search, newTagList);
+  };
+
+  const clearAllTags = () => {
```
Comment (LizardEatsFlies) [inline src/components/local-search/SearchWrapper.tsx:63]: Great Addition!

**Review by LizardEatsFlies — APPROVED** [review]
🚢 it!!!

File: src/components/local-search/SearchWrapper.tsx (hunk 1 above)
Comment (deammer) [reply src/components/local-search/SearchWrapper.tsx:63]: Thank you!

Comment (deammer) [reply src/components/local-search/SearchWrapper.tsx:63]: Thank you!

File: src/components/ToggleFiltersButton.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+const ToggleFiltersButton: FC<Props> = (props) => {
+  const { allActiveTags } = useSearch();
+
+  let label = "";
+  if (allActiveTags.length > 0) {
+    label = allActiveTags.length.toString();
+  }
+  return (
+    <SecondaryButton {...props} className="inline-flex items-center space-x-2">
```
Comment (deammer) [inline src/components/ToggleFiltersButton.tsx:18]: This is a new floating button that toggles the sidebar. It also displays the amount of filters applied to the content of the site.

Screen Shot 2022-03-28 at 5 04 47 PM

File: src/styles/global.css [hunk 3]
```diff
@@ -87,85 +87,114 @@ textarea.input {
...
 .input[type="checkbox"] {
-  @apply mr-2 cursor-pointer border-blue-500;
+  @apply mr-2 cursor-pointer;
+  height: 18px;
+  width: 18px;
 }
 
 /* Projects grid */
+/* 2 cards at 400px + 32px gap = 832px */
+/* 3 cards at 400px + 32px * 2 gap  = 1264px */
```
Comment (deammer) [inline src/styles/global.css:99]: Beware: I did a math. I had to set specific breakpoints to make the content fit within the available space.

File: src/styles/global.css [hunk 4]
```diff
@@ -87,85 +87,114 @@ textarea.input {
...
-@media (min-width: 768px) {
-  .sidebar {
-    position: sticky;
-    top: 70px; /* 56px header height with some buffer */
-    width: 280px;
-    height: calc(100vh - 70px); /* 56px header height with some buffer */
-    z-index: 9999;
-  }
+/* Sidebar with filters */
+.sidebar-backdrop {
```
Comment (deammer) [inline src/styles/global.css:145]: And here's where I styled the sidebar. It's displayed on top of a fixed backdrop with a transparent background, which focuses the eye. I used the `translateX()` directive to make the content slide in from the right side of the screen.

File: src/templates/HomeTemplate.tsx [hunk 5]
```diff
@@ -80,16 +83,21 @@ const HomeTemplate: FunctionComponent<Props> = ({
...
+          <ToggleFiltersButton onClick={() => setShowSidebar(true)} />
+        </div>
+        <div className="mx-auto" style={{ maxWidth: 1600 }}>
           <ProjectList
             allProjects={allProjects.nodes}
             githubDataSet={githubDataSet}
             helpfulnessDataSet={helpfulnessDataSet}
           />
           <SidebarWithFilters
+            showSidebar={showSidebar}
```
Comment (deammer) [inline src/templates/HomeTemplate.tsx:99]: And finally, here's where we dictate whether the sidebar is visible or hidden away.

File: src/components/SidebarWithFilters.tsx [hunk 6]
```diff
@@ -1,39 +1,45 @@
...
 
 type Props = {
   allLanguages: string[];
   allTags: string[];
   allSeeking: string[];
+  showSidebar: boolean;
+  setShowSidebar: Dispatch<SetStateAction<boolean>>;
 };
 
 const SidebarWithFilters: FunctionComponent<Props> = ({
```
Comment (deammer) [inline src/components/SidebarWithFilters.tsx:24]: This is the component that we wanted to modify. Before this PR, it was always visible on the right of the page. With these changes, it'll stay out of the way until users want to filter content.

**Pull request merged**